### PR TITLE
Change the default deadzone value for new actions from 0.5 to 0.2

### LINKED ISSUE #7069: Decrease the default deadzone value for actions to better match real world use cases
### Describe the project you are working on

A game allowing a gamepad as input.

### Describe the problem or limitation you are having in your project

The default deadzone value is `0.5`. By being the default value this implies that it's a reasonable choice.

[Other people recommend a much lower deadzone value (0.05)](https://github.com/godotengine/godot-proposals/issues/3709) and even the [Godot docs use 0.2 as an example value](https://docs.godotengine.org/en/stable/classes/class_inputmap.html#class-inputmap-method-add-action).
According to my experience this also behaves weirdly (on my Xbox 360 controller this feels more like using a d-pad than a stick). This might also confuse newer devs that have never heard about deadzones.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Change the default value to something that is more realistic but also works with older/worse hardware.
Maybe to the `0.2` mentioned in the Godot docs.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Change the default value.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Basically everyone must change the setting in the input map.

### Is there a reason why this should be core and not an add-on in the asset library?

It's already core, it's just about changing the default.

### PR DESCRIPTION
**Pull request by ghost**
This changes the default deadzone value for new actions from 0.5 to 0.2 as discussed in https://github.com/godotengine/godot-proposals/issues/7069.

This does not change the built-in actions.

This is my first pull request, so there is chance I have done something wrong - please check.

* *Bugsquad edit, closes: https://github.com/godotengine/godot-proposals/issues/7069*

### CODE REVIEW

**Comment by AThousandShips** [comment]
You will probably need to add a compatibility note, but the CI will instruct for that

**Comment by AThousandShips** [comment]
Needs a compatibility addition, follow the instructions [here](https://github.com/godotengine/godot/blob/master/misc/extension_api_validation/4.0-stable.expected)

File: misc/extension_api_validation/4.0-stable.expected [hunk 1]
```diff
@@ -433,3 +433,10 @@ GH-79606
 Validate extension JSON: Error: Field 'classes/RenderingDevice/methods/shader_create_from_bytecode/arguments': size changed value in new API, from 1 to 2.
 
 Added optional argument. Compatibility method registered.
+
+
+GH-80299
+--------
+Validate extension JSON: Error: Field 'classes/InputMap/methods/add_action/arguments/1': default_value changed value in new API, from "0.5" to "0.2".
+
+Default deadzone value was changed. No adjustments should be necessary.
```
Comment (AThousandShips) [inline misc/extension_api_validation/4.0-stable.expected:442]: This line should be terminated, add an empty line after

**Comment by AThousandShips** [comment]
Have you made sure this doesn't change input bindings from before this applied? I.e. make a binding with the default deadzone, and then open it with these changes

**Comment by reptofrog** [comment]
> Have you made sure this doesn't change input bindings from before this applied? I.e. make a binding with the default deadzone, and then open it with these changes

Yes, I have tested for that, old actions are not changed.

**Review by Calinou — APPROVED** [review]
Tested locally, it works as expected:

image

This implies a small run-time behavior change, as actions created from a script in `InputMap` will now have a lower deadzone by default. I think this is acceptable (especially since creating new *actions* from code is uncommon), but it needs to be clearly documented in the release notes.

**Comment by Calinou** [comment]
Note that https://github.com/godotengine/godot-demo-projects/pull/1024 changes the deadzone for all non-built-in inputs in demo projects to `0.2`. Merging this PR would ensure we don't accidentally add new actions with a higher deadzone in demo projects :slightly_smiling_face:

**Comment by Sauermann** [comment]
I consider these changes beneficial and think, that they should be included.
Since this PR adjusts the API, perhaps it should be merged early in the 4.4 dev-cycle.

The following two places in the code base should also be investigated, if changes are necessary:

https://github.com/godotengine/godot/blob/7d151c83811f8ac8873439826c16d88c83aba12f/core/config/project_settings.cpp#L506

https://github.com/godotengine/godot/blob/7d151c83811f8ac8873439826c16d88c83aba12f/tests/core/input/test_input_event.h#L93

File: misc/extension_api_validation/4.0-stable.expected
Comment (AThousandShips) [inline misc/extension_api_validation/4.0-stable.expected:1]: Needs to be moved to the 4.3 file when rebasing

**Pull request closed without merging**